lib/modules/ens/index.js

```diff
@@ -361,6 +360,44 @@ class ENS {
     this.embark.addProviderInit('names', code, shouldInit);
     this.embark.addConsoleProviderInit('names', code, shouldInit);
   }
+
+  ensResolve(name, cb) {
+    const self = this;
+    if(!self.configured) {
+      return cb('ENS not configured');
+    }
+    const {ENSRegistry, Resolver} = self.ensConfig;
+    const hashedName = namehash.hash(name);
+    async.waterfall([
+      function getRegistryContract(next) {
+        self.events.request("blockchain:contract:create", {abi: ENSRegistry.abiDefinition, address: ENSRegistry.deployedAddress},
```

Comment:
At the time when this function will be called, `configureContractsAndRegister` will have already be called and it creates the following variables: `self.ensContract`, `self.registrarContract` and `self.resolverContract` that you can use instead of recreating them.

Reply:
I am using the `self.resolverContract` directly. I have a question. Will all the domains have the same resolver as the root domain. If thats the case, then we can use the `self.resolver` directly without getting its address from registry. Please review the new changes